org.civicrm.reqangular/src/common/services/pub-sub-service.js

```diff
@@ -0,0 +1,32 @@
+define([
+  'common/modules/services'
+], function (module) {
+  'use strict';
+
+  var topics = Object.create(null);
+  module.factory("pubSubService", ['$rootScope', function ($rootScope) {
+    return {
+      subscribe: function (topic, listener) {
+        if (!topics[topic]) {
+          topics[topic] = [];
+        }
+        listener.$rootScope = $rootScope;
+        var index = topics[topic].push(listener) - 1;
```

Comment:
Doesn't look nice to have a `var` declaration in the middle of the code implementation. Split things up
```js
function (topic, listener) {
  var index;
  // ....

  index = topics[topic].push(listener) - 1;
```